scripts/publish-npm.js

```diff
@@ -126,12 +126,38 @@ if (nightlyBuild) {
   releaseVersion = tagsWithVersion[tagsWithVersion.length - 1].slice(1);
 }
 
-// -------- Generating Android Artifacts with JavaDoc
-if (exec('./gradlew :ReactAndroid:installArchives').code) {
-  echo('Could not generate artifacts');
-  exit(1);
+const buildAndroid = (rebuildOnError) => {
+  // -------- Generating Android Artifacts with JavaDoc
+  if (exec('./gradlew :ReactAndroid:installArchives').code) {
+    echo('Could not generate artifacts.');
+    exit(1);
+  }
+
+  const aarLocation = `android/com/facebook/react/react-native/${releaseVersion}/react-native-${releaseVersion}.aar`;
+  const tmpLocation = `/tmp/react-native-${releaseVersion}`;
+
+  // -------- Remove unzipped Android artifact (if any) 
+  exec(`rm -rf ${tmpLocation}`);
+
+  // -------- Check if Android artifact contains all the files
+  const {stdout: aarContent} = exec(`unzip ${aarLocation} -d ${tmpLocation} | grep libfbjni.so`);
```

Comment:
`man unzip` says that the `-l` option can just list the contents, which would be a little easier. Also would it make sense to check for other files too?

Reply:
Good idea!

Reply:
Regarding checking other files, I am not sure about that one. In case their number changes, the script will fail. I would assume that if `libfbjni.so` is present, it's fine.